kernelci/build.py

```diff
@@ -321,6 +322,37 @@ def _download_file(url, dest_filename, chunk_size=1024):
     return False
 
 
+def apply_patch_mbox(
+    kdir,
+    mbox_url,
+    git_username="kernelci-tsc",
+    git_email="kernelci-tsc@groups.io"
+):
+    """Download patch mbox from URL and apply with 3-way merge
+
+    *kdir* is the path to a kernel source directory
+    *mbox_url* is the URL to patch mbox content
+    *git_username* is the username used to apply the patch
+    *git_email* is the email used to apply the patch
+    """
+    with tempfile.NamedTemporaryFile(prefix="kernel-patch-") as tmp_f:
+        if not _download_file(mbox_url, tmp_f.name):
+            raise FileNotFoundError(f"Error downloading patch mbox {mbox_url}")
+
+        shell_cmd("""
+set -e
+cd {kdir}
+git config user.name "{git_username}"
+git config user.email "{git_email}"
```

Comment:
No I mean, I think we should actually remove these lines.  Say, if you're doing this over your working kernel source tree and you don't specify command line arguments with your current user name and email then it will replace it.  And then next time you make a commit in the kernel source tree it'll be with the `kernelci-tsc@groups.io` email.

So it should be left to the user to manage this outside of `kci_build`.

Reply:
You mean it should be part of KernelCI config files or something? Sorry, I am not quite following what do you mean. Git won't allow to apply patches without `name` and `email` being set though.

Reply:
It should be part of the user's Git config, yes.  Or we could maybe have a separate command to set the user name and email explicitly if some CI systems need to do that, as a handy wrapper.

> Git won't allow to apply patches without name and email being set though.

Yes and that's fine.  If there's no user name and email configured then git will fail and the issue will be on the user to solve it.  Just like running any other invalid command (say, if `kdir` is not a Git repo etc.).